components/playground/playground.go

```diff
@@ -986,6 +917,103 @@ func (p *Playground) renderSDFile() error {
 	return nil
 }
 
+func (p *Playground) bootMonitor(ctx context.Context, monitorInfo *MonitorInfo) (*exec.Cmd, *grafana, error) {
+	options := p.bootOptions
+
+	// set up prometheus
+	if err := installIfMissing(p.profile, "prometheus", options.version); err != nil {
```

Comment:
It's better to use `PreCommand` too.

Reply:
Prometheus and other components can use `PreCommand`, but grafana can't. It's not a standalone component (we can't run it with `tiup grafana`). Before running it, we need to copy and modify the *.json